core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamUnitTest.java

```diff
@@ -0,0 +1,98 @@
+package com.baeldung.inputstream;
+
+import static org.junit.jupiter.api.io.CleanupMode.ALWAYS;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamUnitTest {
+    @Test
+    public void givenAString_whenWrittenToFileInputStream_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
```

Comment:
Test name does not match what it is doing
```suggestion
    public void givenAString_whenWrittenToFile_thenShouldMatchWhenReadUsingFileInputStream(@TempDir Path tempDir) throws IOException {
```